src/core/config/descriptor_providers/xlio_config_schema.json

```diff
@@ -0,0 +1,1454 @@
+                                    "title": "RX WRE batch size",
+                                    "description": "Number of Work Request Elements and RX buffers to batch before recycling. Maps to XLIO_RX_WRE_BATCHING environment variable. Batching decrease latency mean, but might increase latency STD.\nValue range is 1-1024."
+                                }
+                            }
+                        },
+                        "tcp_segments": {
+                            "type": "object",
+                            "description": "TCP segmentation settings.",
+                            "properties": {
+                                "socket_batch_size": {
+                                    "type": "integer",
+                                    "default": 64,
+                                    "minimum": 1,
+                                    "title": "Socket segment batch size",
+                                    "description": "Number of TCP segments fetched from segments pool by a socket at once. Maps to XLIO_TX_SEGS_BATCH_TCP environment variable. Min value is 1"
+                                },
+                                "ring_batch_size": {
+                                    "type": "integer",
+                                    "default": 1024,
+                                    "minimum": 1,
+                                    "title": "Ring segment batch size",
+                                    "description": "Number of TCP segments fetched from segments pool by a ring at once. Maps to XLIO_TX_SEGS_RING_BATCH_TCP environment variable. Min value is 1"
+                                },
+                                "pool_batch_size": {
+                                    "type": "integer",
+                                    "default": 1024,
+                                    "minimum": 1,
+                                    "title": "Pool segment batch size",
+                                    "description": "Number of TCP segments batched when fetched from the segments pool. Maps to XLIO_TX_SEGS_POOL_BATCH_TCP environment variable. Min value is 1"
+                                }
+                            }
+                        }
+                    }
+                },
+                "max_gro_streams": {
+                    "type": "integer",
+                    "default": 32,
+                    "title": "Maximum GRO streams",
+                    "description": "Control the number of TCP streams to perform Generic Receive Offload simultaneously. Maps to XLIO_GRO_STREAMS_MAX environment variable. Disable GRO with a value of 0."
+                }
+            }
+        },
+        "applications": {
+            "type": "object",
+            "title": "Application Integration",
+            "description": "Configurations and optimizations for specific applications or use cases.",
+            "properties": {
+                "nginx": {
```

Comment:
We have also envoy support if we want to consider it.  afair some params are common ...

Reply:
I don't want to call it nginx_envoy or something like that - 
I can duplicate this group of options to envoy, but maybe we'll get to the bridge when we'll need envoy?